file-formats/nrob/type/zif_nr_aff_nrob_v1.intf.abap

```diff
@@ -0,0 +1,117 @@
+      prefix               TYPE abap_bool,
+    END OF ty_interval.
+
+
+  "! <p class="shorttext">Buffering</p>
+  "! Choose a buffer type for no buffering, for buffering via main memory or for parallel buffering.
+  "! $values {@link zif_nr_aff_nrob_v1.data:co_buffering}
+  TYPES ty_buffering TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Buffering</p>
+    "! Choose a buffer type for no buffering, for buffering via main memory or for parallel buffering.
+    BEGIN OF co_buffering,
+      "! <p class="shorttext">Main Buffer</p>
+      "! Main memory buffering. \n\n Main memory buffering is the most efficient buffering solution with regards
+      "! to system performance. Almost all number range objects use main memory buffering. However, it does have
+      "! two restrictions: Numbers can no longer be assigned in chronological order. The numbering may contain gaps.
+      main_buffer TYPE ty_buffering VALUE 'X',
+      "! <p class="shorttext">Parallel</p>
+      "! Parallel buffering, \n\n Parallel buffering works in the same way as main memory buffering, with the
+      "! difference that the number packages are not stored in the main memory, but are written to the
+      "! table NRIVSHADOW.
+      parallel    TYPE ty_buffering VALUE 'S',
+      "! <p class="shorttext">None</p>
+      "! No buffering \n\n A number is taken from an interval stored in database NRIV. The table line is
+      "! locked and this lock remains until the application that requested the number is closed, either
+      "! with a COMMIT WORK or a ROLLBACK.
+      none        TYPE ty_buffering VALUE ' ',
+    END OF co_buffering.
+
+  TYPES:
+    "! <p class="shorttext">Configuration</p>
+    "! Configuration-specific fields
+    "! $required
+    BEGIN OF ty_configuration,
+      "! <p class="shorttext">Transaction ID</p>
+      "! Transaction code for application specific transaction
+      transaction_id   TYPE c LENGTH 20,
+      "! <p class="shorttext">Buffering</p>
+      "! Choose a buffer type for no buffering, for buffering via main memory or for parallel buffering.
+      "! Default is mainBuffer.
+      "! $required
+      buffering        TYPE ty_buffering,
+      "! <p class="shorttext">Buffered Numbers</p>
+      "! This value specifies the numbers in buffer. In case of parallel and main memory buffering, add a
+      "! number for \u0027bufferedNumbers\u0027. It determines how many numbers are reserved in buffer for
+      "!  the intervals.Default number of buffers is 10.
+      "! $required
+      buffered_numbers TYPE n LENGTH 8,
+    END OF ty_configuration.
+
+  TYPES:
+    "! <p class="shorttext">NROB Object Type</p>
+    "! Object type NROB
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Schema</p>
+      "! Format version
+      "! $required
+      schema        TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header        TYPE if_aff_types_v1=>ty_header_60,
```

Comment:
```suggestion
      header        TYPE zif_aff_types_v1=>ty_header_60,
```

Reply:
Yap, definitely need more automatization. Manual renaming is error prone